ros2/test/pluginlib/BUILD.bazel

```diff
@@ -13,8 +14,8 @@ ros2_plugin(
     srcs = ["square.cc"],
     plugin_specs = [
         {
-            "class_type": "polygon_plugins::Square",
```

Comment:
I don't see how are those namespacing changes relevant to this PR. Please revert. We can discuss this in a separate PRs. Your tests should work with the existing plugins AFAICS.

Reply:
I'm still looking for the bit of code in the pluginlib, classloader and ament code that makes these changes required. Somewhere in that code an assumption seems to be made that all classes in a single namespace are exported by the same plugin. The code doesn't work without these changes. I'm happy to extract them into another PR as well if you prefer.

Reply:
Why would we need those changes in this PR except for testing? Can you refactor the tests to work with the current test code?

Reply:
OK, fair enough; it's strange this doesn't work btw (have no idea whether this is supposed to work w/ regular ROS workspaces). Anyway, you can create another set of simple test plugins, e.g. use a genrule to generate source files and plugin targets using list comprehension in the build file. The existing tests and the new ones do not test the same functionality, they are unrelated IMO.

Reply:
I looked a bit further and the reason it wouldn't work is that the generated package name of both the `triangle` and `square` plugins are the same (because of the way `_get_package_name` is implemented in `ament.bzl`. I guess in the normal colcon ros2 world these are manually specified by the user. I don't think we could change this logic without breaking existing code, but at least now we know why it wouldn't work before. I will create some alternative tests for my PR instead as you suggested.

Reply:
It might be that indeed that use-case is not covered there. We can refactor/upgrade the relevant code when needed.